src/components/movies-list/movies-list.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { MovieCard } from '../movie-card/movie-card';
+import VisibilityFilterInput from '../visibility-filter-input/visibility-filter-input';
+
+const mapStateToProps = (state) => {
+	const { visibilityFilter } = state;
+	return { visibilityFilter };
+};
+
+function MoviesList(props) {
+	const { movies, visibilityFilter } = props;
+	let filteredMovies = movies;
+
+	if (visibilityFilter !== '') {
+		filteredMovies = movies.filter((m) => m.Title.toLowerCase().includes(visibilityFilter.toLowerCase()));
+	}
+
+	if (!movies) return <div className="main-view" />;
+
+	return (
+		<div className="movies-list">
+			<VisibilityFilterInput visibilityFilter={visibilityFilter} />
+			{filteredMovies.map((m) => (
+				<MovieCard key={m._id} movie={m} />
+			))}
+		</div>
+	);
+}
+
```

Comment:
if going to be better add `mapStateToProps` and `mapdispatchtoprops` here. writing this code above the connect provide better Readability

Reply:
This has been updated.